### ISSUE TITLE
[Help Wanted] Add new messages indicator when scrolled up in a channel

### ISSUE THREAD

**Issue by jasonblais** [mattermost/platform#4250]
We're looking for anyone interested in helping contribute this improvement.

If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team.

---

When a user scrolls up in a channel and a new message appears below, there is no visual cue for the user that a new message has arrived.

This improvement adds a new messages indicator to show that there is a new message at the bottom of the channel (see screenshot). The indicator should disappear when the user scrolls down to the bottom of the channel. ([Jira ticket](https://mattermost.atlassian.net/browse/PLT-1917))

Note: If you need any help with CSS, feel free to submit the pull request and we're happy to help tweak the indicator if needed.

![image](https://cloud.githubusercontent.com/assets/13119842/19465047/10444d58-94cf-11e6-823f-b63bad3b685f.png)

![image](https://cloud.githubusercontent.com/assets/13119842/19465067/2af3e2b2-94cf-11e6-8b3c-99bc3ffb193b.png)

**Comment by isundil** [comment]
Why not having a "new message notification" on the chan list bar, like in this picture ?

https://www.mattermost.org/wp-content/uploads/2015/06/20150715_non-english-channels4.jpg

**Comment by pepf** [comment]
I can work on this! Is the icon from the indicator already available?

@isundil The notifications from the picture you linked to are for specific @mentions in a channel as far as ( can see ). This indicator will show _all_ unread messages in a specific channel. Right?

**Comment by jasonblais** [comment]
That's correct @pepf, the message indicator will show how many new messages appeared.

I'll also get you the icon (we should have that available) or else let you know what existing one to use. Sorry for the omission.

PS: I added a note that if the user has scrolled to the bottom (and hence seen all the messages), this indicator should not appear.

**Comment by asaadmahmood** [comment]
We're using the icon: http://fontawesome.io/icon/arrow-circle-o-down/

**Comment by pepf** [comment]
@asaadmahmood Should the grey of the indicator correspond with a specific theme color or always be the same color as in the image? ( #9ca3ac )

**Comment by asaadmahmood** [comment]
@pepf Instead of going with the layout attached. Just use the link color as the background, and the center channel background as the text color.

**Comment by pepf** [comment]
I prepared a PR (#4299) for this issue. It works as intended but because of the _new messages_ header it can potentially cause some confusion in the way new messages are referred to. To clarify this a bit further I added a gif here:

![newmessage](https://cloud.githubusercontent.com/assets/1265435/19628731/a958d36c-9964-11e6-9032-5066c1deed83.gif)

I looked into the way this header behaves functionally and found it quite complex to quickly integrate it into this feature as well. It is probably best to create a new ticket for this...
